rcsb/utils/dictionary/DictMethodAssemblyHelper.py

```diff
@@ -223,7 +223,7 @@ def buildContainerAssemblyIds(self, dataContainer, catName, **kwargs):
 
     def addDepositedAssembly(self, dataContainer, catName, **kwargs):
```

Comment:
@piehld Would it be better to create a new method `addCompModelDepositedAssembly` and use that in the RCSB extension dictionary for computed models? This method, `addDepositedAssembly`, can be cleaned up to be applied for experimental models, and suppressed at the dictionary level. This would allow more flexibility.

Reply:
@brindakv My understanding from the production level dictionary is that `addDepositedAssembly` was commented out for experimental models—doesn't that mean the method wasn't running at all for experimental models? If my understanding is correct, then we wouldn't need to create two separate methods. (Although, I would agree that renaming it to `addCompModelDepositedAssembly` would be appropriate, to make it clear that it only runs for computational models.)

Reply:
Although `addDepositedAssembly` was commented out in the dictionary for experimental models, the same method is applied for computed models. Note that method invocation is initiated based on the dictionary definitions and not based on the data instance (i.e., not based on whether the data instance is a computed model or an experimental model but based on what is defined in the dictionary; hence the separate dictionaries to begin with). When we merged the dictionaries, this allowed for the method to be applied for both types of models because it was present in the dictionary and was being called regardless of the model type. The code here also allowed for implementing the method for both computed and experimental models based. Now that the dictionaries are merged, one way to suppress methods for specific types of models, is to create completely different methods for experimental and computed models.

Reply:
I understand that the effect of merging the dictionaries meant that this method would now be applied to both experimental and computational models, but if the current production pipeline isn't running that method on experimental models (as it is commented out in the `rcsb_mmcif_ext_v1.dic` [dictionary that's being used in production](https://raw.githubusercontent.com/rcsb/py-rcsb_exdb_assets/master/dictionaries/rcsb_mmcif_ext_v1.dic)), then why would we run it on experimental models in the future? We're no longer including deposited assemblies for experimental models.

Reply:
It allows flexibility for the future in case requirements change. And retains control at the dictionary level.

Reply:
OK, I see your point. In that case though, I'm actually more inclined to keep it as a single combined method, since no matter what (whether it's a combined method or two separate methods) we would need to include a conditional to determine if the given model is experimental or computational. 

If we envision some of the existing code for the method being applied to experimental models, then perhaps it makes more sense to keep just the one method, with a single conditional checking if the given model is exp. or comp. (as was basically being done up to now). But, if we envision the method looking completely different for experimental models, then it probably makes more sense to separate it into two different methods, each with a conditional at the beginning checking whether the given model is exp. or comp. 

For now, I will plan to move this method to the `DictMethodCompModelHelper` class (leaving the same method name though since it'll be in a comp. model-specific class). Then if we later want to add similar details for experimental models, we can move or copy this method back to this `AssemblyHelper` class and decide on the best way to proceed then.

Does that plan sound OK to you?

Reply:
@piehld it is perhaps better to set up a meeting to discuss this issue.

Reply:
OK sure, that would be fine. I don't feel strongly one way or the other. The only thing I'd prefer to avoid is keeping an unused method in the codebase, which may lead to confusion later on (as it already did in this situation here).